test-harness/modules/mocker/endpoints/marathon.py

```diff
@@ -372,15 +375,15 @@ class MarathonEndpoint(RecordingTcpIpEndpoint):
     """An endpoint that mimics DC/OS root Marathon"""
     def __init__(self, port, ip=''):
         super().__init__(port, ip, MarathonHTTPRequestHandler)
-        self._context.data["endpoint-content"] = {"apps": [NGINX_TASK_ALWAYSTHERE, ]}
```

Comment:
Doing it this way will result in confusing logging messages e.g. here https://github.com/dcos/adminrouter/blob/master/test-harness/modules/mocker/endpoints/generic.py#L127 - `init ~= reset`.

For me, an ideal solution would be to have a separate function (e.g. `_set_data` or something) that will be called by both `__init__` and `reset`. If you are up to a task, we can refactor it now, but I would like us to be consistent and change *all* the Endpoint-derived classes to follow that pattern. If you would like to keep this task/PR short - let's please do not modify it in just one place - test harness is already very complex, I fear that if we do not keep it very tidy, it will quickly become difficult to use and maintain.